content/diferenca-operadores.md

```diff
@@ -0,0 +1,108 @@
+Title: Diferença entre == e is
+Slug: diferenca-operadores
+Date: 2015-11-27 14:00
+Tags: python,dica,operadores
+Author: Gildásio Júnior
+Email:  gjuniioor@protonmail.ch
+Github: gjuniioor
+Site: https://gjuniioor.github.io
+Category: Python
+
+Diferença entre == e is
+-----------
+
+Olá, galera, tudo tranquilo?
+
+Estava conversando com um amigo que está estudando python e tudo mais, e então ele veio com a seguinte dúvida:
+
+> Qual a diferença entre == e is no python?
+
+Para quem não sabe, no python tem o operador *is* que "tem a mesma função do *==*. Veja:
```

Comment:
essa parte está confusa porque o operador `is` NÃO tem a mesma função do operador `==`.

`==` verifica se um objeto A e um objeto B são [equivalentes](https://docs.python.org/3/library/operator.html#operator.__gt__). 

`is` verifica se a identidade de um objeto é a mesma, se é exatamente o mesmo objeto em memoria.

podem existir dois objetos, um A e um B que são equivalentes mas são dois objetos diferentes em memoria.

isso fica mais explicito utilizando classes como exemplo:

```python
>>> class Pessoa(object):
...     def __init__(self, x):
...         self.x = x
...     
...     def __eq__(self, other):
...         return self.x == other.x
... 
>>> a = Pessoa(2)
>>> b = Pessoa(2)
>>> a == b
True
>>> a is b
False
>>> 

```

http://stackoverflow.com/questions/132988/is-there-a-difference-between-and-is-in-python